ws-client/src/transport.rs

```diff
@@ -71,9 +70,9 @@ pub struct WsTransportClientBuilder<'a> {
 	pub target: Target,
 	/// Timeout for the connection.
 	pub timeout: Duration,
-	/// `Origin` header to pass during the HTTP handshake. If `None`, no
-	/// `Origin` header is passed.
-	pub origin_header: Option<Cow<'a, str>>,
+	/// Custom headers to pass during the HTTP handshake. If `None`, no
+	/// custom header is passed.
+	pub custom_headers: Vec<Header<'a>>,
```

Comment:
```suggestion
	/// Headers passed to the server during the HTTP handshake.
	pub headers: Vec<Header<'a>>,
```

Could it be a slice, `&'a [Header<'a>]`, instead of a `Vec` you think?

Reply:
it could be but then we need to have `add_headers(&[&str, &str])` instead that it slightly more error prone I think but would work...

Reply:
when we can avoid allocations without sacrificing "ergonomics" then it's a go but we should prioritize "ergonomics" especially in "builders" that only performed once.

but we have no dedicated rules rather than that @maciejhirsz likes to remove needless allocations :)